File: Exo MattieuGarcia/Rencontre.php
<?php
require_once 'Personnage.php';

class Rencontre
{
    private $critRate;
    private $successRate;
    private $failureRate;
    private $fumbleRate;
    private $baseVitalityLoss;
    private $bonusEquipment;

    public function __construct($critRate, $successRate, $failureRate, $fumbleRate, $baseVitalityLoss, $bonusEquipment)
    {
        $this->critRate = $critRate;
        $this->successRate = $successRate;
        $this->failureRate = $failureRate;
        $this->fumbleRate = $fumbleRate;
        $this->baseVitalityLoss = $baseVitalityLoss;
        $this->bonusEquipment = $bonusEquipment;
    }

    public function calculateRates(Personnage $personnage)
    {
        // Apply modifiers based on character's stats
        $modifiers = [];

        // Example modifiers based on character's stats
        if ($personnage->getRapidite() <= 5) {
            $modifiers['fumble'] = 5;
        }

        if ($personnage->getForce() >= 10) {
            $modifiers['crit'] = 5;
        }

        // Apply modifiers to rates
        $critRate = $this->critRate + ($modifiers['crit'] ?? 0);
        $successRate = $this->successRate;
        $failureRate = $this->failureRate;
        $fumbleRate = $this->fumbleRate + ($modifiers['fumble'] ?? 0);

        return [
            'crit' => $critRate,
            'success' => $successRate,
            'failure' => $failureRate,
            'fumble' => $fumbleRate
        ];
    }

    // Getter methods for other properties can be added if needed
}

File: POO_PYTHON/Rencontre.py
class Rencontre:
    def __init__(self, crit_rate, success_rate, failure_rate, fumble_rate, perte_vitalite, equipement_bonus):
        self.crit_rate = crit_rate
        self.success_rate = success_rate
        self.failure_rate = failure_rate
        self.fumble_rate = fumble_rate
        self.perte_vitalite = perte_vitalite
        self.equipement_bonus = equipement_bonus

    def get_rates(self, personnage):
        rates = {
            "Critique": self.crit_rate,
            "Succes": self.success_rate,
            "Echec": self.failure_rate,
            "Fumble": self.fumble_rate
        }
        if personnage.rapidite <= 5:
            rates["Fumble"] += 5
        if personnage.force >= 10:
            rates["Critique"] += 5
        return rates
